Configs/.local/share/bin/cliphist.sh

```diff
@@ -1,86 +1,90 @@
 #!/usr/bin/env sh
 
-scrDir=`dirname "$(realpath "$0")"`
+scrDir=$(dirname "$(realpath "$0")")
 source $scrDir/globalcontrol.sh
-roconf="~/.config/rofi/clipboard.rasi"
+roconf="${confDir}/rofi/clipboard.rasi"
 
+#? Cursor position offset when pastebin is spawned
+x_offset=15  #* Cursor spawn position on clipboard
+y_offset=210 #* To point the Cursor to the 1st and 2nd latest word
 
-# set position
-x_offset=-15   #* Cursor spawn position on clipboard
-y_offset=210   #* To point the Cursor to the 1st and 2nd latest word
-#!base on $HOME/.config/rofi/clipboard.rasi 
-clip_h=$(cat "${confDir}/rofi/clipboard.rasi" | awk '/window {/,/}/'  | awk '/height:/ {print $2}' | awk -F "%" '{print $1}')
-clip_w=$(cat "${confDir}/rofi/clipboard.rasi" | awk '/window {/,/}/'  | awk '/width:/ {print $2}' | awk -F "%" '{print $1}')
-#clip_h=55 #! Modify limits for size of the Clipboard
-#clip_w=20 #! This values are transformed per cent(100)
-#? Monitor resolution , scale and rotation 
-x_mon=$(hyprctl -j monitors | jq '.[] | select(.focused==true) | .width')
-y_mon=$(hyprctl -j monitors | jq '.[] | select(.focused==true) | .height')
-#? Rotated monitor? 
-monitor_rot=$(hyprctl -j monitors | jq '.[] | select(.focused==true) | .transform')
-if [ "$monitor_rot" == "1" ] || [ "$monitor_rot" == "3" ]; then  # if rotated 270 deg
- tempmon=$x_mon
-    x_mon=$y_mon
-    y_mon=$tempmon
-#! For rotated monitors
-fi
-#? Scaled monitor Size
-monitor_scale=$(hyprctl -j monitors | jq '.[] | select (.focused == true) | .scale' | sed 's/\.//')
-x_mon=$((x_mon * 100 / monitor_scale ))
-y_mon=$((y_mon * 100 / monitor_scale))
-#? monitor position
-x_pos=$(hyprctl -j monitors | jq '.[] | select(.focused==true) | .x')
-y_pos=$(hyprctl -j monitors | jq '.[] | select(.focused==true) | .y')
-#? cursor position
-x_cur=$(hyprctl -j cursorpos | jq '.x')
-y_cur=$(hyprctl -j cursorpos | jq '.y')
-# Ignore position
- x_cur=$(( x_cur - x_pos))
- y_cur=$(( y_cur - y_pos))
-#Limiting
-# Multiply before dividing to avoid losing precision due to integer division
-clip_w=$(( x_mon*clip_w/100 ))
-clip_h=$(( y_mon*clip_h/100 ))
-max_x=$((x_mon - clip_w - 5 )) #offset of 5 for gaps
-max_y=$((y_mon - clip_h - 15 )) #offset of 15 for gaps
-x_cur=$((x_cur - x_offset))
-y_cur=$((y_cur - y_offset))
-# 
-x_cur=$(( x_cur < min_x ? min_x : ( x_cur > max_x ? max_x :  x_cur)))
-y_cur=$(( y_cur < min_y ? min_y : ( y_cur > max_y ? max_y :  y_cur)))
+#? Parse clipboard.rasi and fetch the width. Should consider percent
+clpWidth=$(awk '/window {/,/}/ { if (/width:/) { split($2, a, "%"); print a[1] } }' "${roconf}" )
```

Comment:
I still don't remember what i was on, to set window width in %  :)
I think it would be better to use `em` unit for width and height.

Reply:
Okie, As I have understood this, if we set it to em it will scale to the set font size right?

Reply:
> Okie, As I have understood this, if we set it to em it will scale to the set font size right?

yup, setting it to em will scale the menu based on font size.. so for example, if we set font size as `10` and width as `30em` then it will scale to `300px` on screen... so we dont care about the monitor specs...

Reply:
Well, try to adjust on how this treats the boundaries. But Goodnight for now 😂

Reply:
> Well, try to adjust on how this treats the boundaries. But Goodnight for now 😂

once you have fixed size of rofi window... it should easy to do that, like windows does... take the cursor position and check the below and above... and offset the rofi window accordingly

Reply:
So what happens now is I got unknown issues. Does rofi adheres to the em unit * fontsize ?

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/3e911c17-022f-44b2-baf4-7ee5a500a51c)


So I'm using the default fnt size : 9 and 20em for W and 30 em for H
So hyprctl reports that the rofi layer if we try to see the ratio uses 8.5 as fnt size.  which ruins the boundaries. ( lazy to debug which part the .rasi does this but I am certain the scripts follows the em * fnt  principle )


For now, to solve this I still end up overwriting the width and height.

Will push this later for review and revision

Reply:
I also tried to get the equivalents in percent.